New course colors

firejake308 opened the pull request:
## Description

The current system renders all meetings on the schedule using variants of maroon. This is aesthetically consistent, but makes it difficult to quickly differentiate between different classes. Therefore, we are considering a new color scheme with more diverse colors. By keeping all of the colors on the dark side, we should be able to maintain aesthetic consistency.

## Rationale

So glad I made that `useMeetingColors` hook now

## How to test

Add like 10 courses to your schedule

## Screenshots

|Before|After|
|--|--|
|![image](https://user-images.githubusercontent.com/10082177/98322249-9cc62780-1fac-11eb-9b2f-71c65016dd45.png)|![image](https://user-images.githubusercontent.com/10082177/98322161-65f01180-1fac-11eb-9c14-6aec383f4637.png)|

## Related tasks

Closes #374

gannonprudhomme commented:
So I used this website: https://www.color-blindness.com/coblis-color-blindness-simulator/ to simulate color blindness (you can read about the different types [here](https://www.nei.nih.gov/learn-about-eye-health/eye-conditions-and-diseases/color-blindness/types-color-blindness)) and it seems like these two pairs are gonna be the hardest to differentiate:
|A|B|
|--|--|
|![chrome_lypMJWF0N8](https://user-images.githubusercontent.com/7295783/98411843-6506c080-203c-11eb-8c03-cd6749e286c8.png)|![chrome_ho4wjOGchv](https://user-images.githubusercontent.com/7295783/98411844-659f5700-203c-11eb-96bf-d0ebfeb68290.png)|

gannonprudhomme commented:
Also while I do like having the color order be like a gradient, it'll probably be best to have the colors that are alike to be farther apart from each other (basically mix them up so it's not a gradient).

firejake308 commented:
So the red and purple honestly looked kind of similar if I just focused away from the screen/zoomed out, and my solution was to pump up the amount of red in the red and decrease the amount of green and blue channels. I didn't want to decrease the blue channel alone, because that made it too similar to orange.

As for the orange and the light brown, they look pretty different to me with full-color vision, but it seems that under red-blind and green-blind conditions, they look essentially the same. I guess that without red or green cones, you'd be relying disproprtionately on the blue channel, which is similar for both. I tried increasing the amount of blue in the light brown, which made it turn sort of pink-ish, but it was still differentiable from all other colors and also differentiable under color-blind conditions.

Screenshot of all colors:
![image](https://user-images.githubusercontent.com/10082177/99753567-0580ca00-2aac-11eb-918d-ef0628356e3d.png)

gannonprudhomme reviewed (changes requested):
From the meeting, we said to replace the tan / "light brown/pink" to something else

firejake308 commented:
Changed tan to pink:
|Old|New|
|--|--|
|![image](https://user-images.githubusercontent.com/10082177/101999695-4c29a480-3ca5-11eb-99d2-9587291fef92.png)|![image](https://user-images.githubusercontent.com/10082177/101999719-74190800-3ca5-11eb-9e24-9c8d249ca844.png)|

gannonprudhomme commented:
Just need to reorganize them and we'll be good

firejake308 commented:
Renamed tan to yellow, moved dark brown to top and red to bottom

![image](https://user-images.githubusercontent.com/10082177/103181327-92663280-4865-11eb-891b-dd61762f3abe.png)

gannonprudhomme commented on autoscheduler/frontend/src/components/SchedulingPage/Schedule/meetingColors.ts:
```diff
@@ -2,8 +2,15 @@ import { useSelector } from 'react-redux';
 export const colors = [
-  '#500000', '#733333', '#966666', '#b99999',
-  '#871b1e', '#9a1d26', '#c2777d', '#9f494b', '#b76778',
+  '#f2c05b', // yellow
+  '#612b0d', // dark brown
+  '#2d683e', // green
+  '#ec4631', // orange
+  '#227d9c', // blue
+  '#83205c', // purple
+  '#db728e', // light brown/pink
+  '#249889', // turquoise
+  '#c82041', // red
```
```suggestion
   '#f2c05b', // yellow
  '#2d683e', // green
  '#ec4631', // orange
  '#227d9c', // blue
  '#83205c', // purple
  '#db728e', // light brown/pink
  '#249889', // turquoise
  '#c82041', // red
  '#612b0d', // dark brown
```
Moves brown to be the last one

firejake308 commented:
Final screenshot:
![image](https://user-images.githubusercontent.com/10082177/103185051-3fe63f80-4880-11eb-9837-33b22642c243.png)
